awscli/customizations/codeartifact/login.py

```diff
@@ -84,13 +84,16 @@ def _run_commands(self, tool, commands, dry_run=False):
 
     def _run_command(self, tool, command, *, ignore_errors=False):
         try:
-            self.subprocess_utils.check_call(
+            self.subprocess_utils.run(
                 command,
-                stdout=self.subprocess_utils.PIPE,
-                stderr=self.subprocess_utils.PIPE
+                capture_output=True,
+                check=True
             )
         except subprocess.CalledProcessError as ex:
             if not ignore_errors:
+                sys.stdout.write(
```

Comment:
Instead of writing out before raising the called process error, it would be interesting if we can explore just exposing our own exception class that we can raise and control the message to include stderr for. We can the exception something like `CommandFailedError` that accepts the instance of a `CalledProcessError` and we can then format the error message to include the normal string repr for `CalledProcessError` and then whatever was displayed from `stderr` to make the message like:
```
Command '['some-command', 'login']' returned non-zero exit status 1. Stderr from command: <stderr content>
```
I like this approach more as:
1. It give a little more context on what command error output represents
2. It guarantees we are writing this error message to stderr by including it in the raised exception. In general, we only want to be including error messages in stderr and not stdout as that is a general expectation from the user.

Reply:
I created a class that inherits `CalledProcessError`. For the most part they are the same, other than when returncode > 1 (exception raised) it will add the info from stderr.